rust-toolchain.toml

```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "nightly-2022-07-22"
+channel = "nightly-2023-01-04"
```

Comment:
Bumped the nightly to get `error_in_core`, looks like we missed it by a week from previous nightly, jumped to this version to match version at https://github.com/mobilecoinfoundation/mobilecoin/pull/3020

Reply:
no longer relevant, change was made in #31